src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -49,37 +55,60 @@ public ElasticsearchCaseSearchOperation(JestClient jestClient,
     }
 
     @Override
-    public CaseSearchResult execute(CaseSearchRequest caseSearchRequest) {
-        SearchResult result = search(caseSearchRequest);
+    public CaseSearchResult execute(CrossCaseTypeSearchRequest request) {
+        MultiSearchResult result = search(request);
         if (result.isSucceeded()) {
             return toCaseDetailsSearchResult(result);
         } else {
             throw new BadSearchRequest(result.getErrorMessage());
         }
     }
 
-    private SearchResult search(CaseSearchRequest caseSearchRequest) {
-        Search searchRequest = secureAndTransformSearchRequest(caseSearchRequest);
+    private MultiSearchResult search(CrossCaseTypeSearchRequest request) {
+        MultiSearch multiSearchAction = secureAndTransformSearchRequest(request);
         try {
-            return jestClient.execute(searchRequest);
+            return jestClient.execute(multiSearchAction);
         } catch (IOException e) {
             throw new ServiceException("Exception executing Elasticsearch : " + e.getMessage(), e);
         }
     }
 
-    private Search secureAndTransformSearchRequest(CaseSearchRequest caseSearchRequest) {
-        CaseSearchRequest securedSearchRequest = caseSearchRequestSecurity.createSecuredSearchRequest(caseSearchRequest);
+    private MultiSearch secureAndTransformSearchRequest(CrossCaseTypeSearchRequest request) {
+        Collection<Search> securedSearchActions = request.getCaseTypeIds()
+            .stream()
+            .map(caseTypeId -> createSecuredSearchAction(caseTypeId, request.getSearchRequestJsonNode()))
+            .collect(toList());
+
+        return new MultiSearch.Builder(securedSearchActions).build();
+    }
+
+    private Search createSecuredSearchAction(String caseTypeId, JsonNode searchRequestJsonNode) {
+        CaseSearchRequest securedSearchRequest = caseSearchRequestSecurity.createSecuredSearchRequest(new CaseSearchRequest(caseTypeId, searchRequestJsonNode));
         return new Search.Builder(securedSearchRequest.toJsonString())
-            .addIndex(getCaseIndexName(caseSearchRequest.getCaseTypeId()))
+            .addIndex(getCaseIndexName(caseTypeId))
             .addType(getCaseIndexType())
             .build();
     }
 
-    private CaseSearchResult toCaseDetailsSearchResult(SearchResult result) {
-        List<String> casesAsString = result.getSourceAsStringList();
+    private CaseSearchResult toCaseDetailsSearchResult(MultiSearchResult multiSearchResult) {
+        List<CaseDetails> caseDetails = new ArrayList<>();
+        AtomicLong totalHits = new AtomicLong(0);
+
+        multiSearchResult.getResponses()
+            .stream()
+            .filter(response -> response.searchResult != null)
+            .forEach(response -> {
+                caseDetails.addAll(searchResultToCaseList(response.searchResult));
```

Comment:
@gbenadikar have you checked with Steve robinson and Valentin whether it's ok to merge all the results in one list or perhaps is better for the clients to have them in a Map grouped by case type id?
Same for the total hits. 
Let's clarify this now and avoid having to change this in the future

Reply:
The last task RDM-3487 enforces the multi search results to return uniform JSON structure across multiple case types. The results will have to be normalised and only common fields will be returned (metadata and search alias fields). So this implementation is as expected.